### PR TITLE
wq: add input and output lines in the transaction log

### PR DESCRIPTION
**Pull request by sghuang19**
I'm not sure how to test it, but the codes are done.

### CODE REVIEW

**Comment by sghuang19** [comment]
#2879

File: work_queue/src/work_queue.c [hunk 1]
```diff
@@ -1403,6 +1404,8 @@ static work_queue_result_code_t get_output_file( struct work_queue *q, struct wo
 		w->total_transfer_time += sum_time;
 
 		debug(D_WQ, "%s (%s) sent %.2lf MB in %.02lfs (%.02lfs MB/s) average %.02lfs MB/s", w->hostname, w->addrport, total_bytes / 1000000.0, sum_time / 1000000.0, (double) total_bytes / sum_time, (double) w->total_bytes_transferred / w->total_transfer_time);
+
+        write_transaction_transfer(q, w, t, total_bytes, sum_time, f->remote_name, 0);
```
Comment (btovar) [inline work_queue/src/work_queue.c:1408]: Instead of 0, let's use:  `WORK_QUEUE_OUTPUT`

File: work_queue/src/work_queue.c [hunk 2]
```diff
@@ -3386,6 +3389,9 @@ static work_queue_result_code_t send_input_file(struct work_queue *q, struct wor
 
 		q->stats->bytes_sent += total_bytes;
 
+		// Write to the transaction log.
+		write_transaction_transfer(q, w, t, total_bytes, elapsed_time, f->remote_name, 1);
```
Comment (btovar) [inline work_queue/src/work_queue.c:3393]: As above, change 1 to `WORK_QUEUE_INPUT`

File: work_queue/src/work_queue.c [hunk 3]
```diff
@@ -7543,6 +7549,25 @@ static void write_transaction_worker_resources(struct work_queue *q, struct work
 }
 
 
+static void write_transaction_transfer(struct work_queue *q, struct work_queue_worker *w, struct work_queue_task *t, size_t sizeinmb, int walltime, char* filename, int type){
+	struct buffer B;
+	buffer_init(&B);
+	buffer_printf(&B, "TRANSFER ");
+	if (type)
```
Comment (btovar) [inline work_queue/src/work_queue.c:7556]: Use braces for both the `if` and `else`, even it it is only one statement.
Also, now use
```
if(type == WORK_QUEUE_INPUT) {  
    ...
```

File: work_queue/src/work_queue.c [hunk 4]
```diff
@@ -7543,6 +7549,25 @@ static void write_transaction_worker_resources(struct work_queue *q, struct work
 }
 
 
+static void write_transaction_transfer(struct work_queue *q, struct work_queue_worker *w, struct work_queue_task *t, size_t sizeinmb, int walltime, char* filename, int type){
```
Comment (btovar) [inline work_queue/src/work_queue.c:7552]: Note that size and walltime come in bytes, and microseconds, respectively.  I'd suggest changing the parameters to:
```C
..., size_t size_in_bytes, int time_in_usecs, work_queue_file_type_t type) {
```

and convert to MB and seconds as needed. Use "%f" when printing the numbers below.

**Review by btovar — CHANGES_REQUESTED** [review]

File: work_queue/src/work_queue.c (hunk 4 above)
Comment (btovar) [reply work_queue/src/work_queue.c]: For memory, divide by `((double) MEGABYTE)`.  For time, divide by `((double) USECOND)`, and use "%f" in buffer_print for both.

**Review by btovar — APPROVED** [review]

**Pull request merged**